service/history/workflow_rebuilder.go

```diff
@@ -77,47 +88,218 @@ func NewWorkflowRebuilder(
 func (r *workflowRebuilderImpl) rebuild(
 	ctx context.Context,
 	workflowKey definition.WorkflowKey,
+	branchToken []byte,
 ) (retError error) {
+	if len(branchToken) != 0 {
+		return r.rebuildFromBranchToken(ctx, workflowKey, branchToken)
+	}
+	return r.rebuildFromMutableState(ctx, workflowKey)
+}
 
-	wfContext, err := r.workflowConsistencyChecker.GetWorkflowContext(
+func (r *workflowRebuilderImpl) rebuildFromBranchToken(
+	ctx context.Context,
+	workflowKey definition.WorkflowKey,
+	branchToken []byte,
+) (retError error) {
+	if workflowKey.RunID == "" {
+		return serviceerror.NewInvalidArgument("unable to rebuild mutable state using branch token without run ID")
+	}
+
+	wfCache := r.workflowConsistencyChecker.GetWorkflowCache()
+	wfContext, releaseFn, err := wfCache.GetOrCreateWorkflowExecution(
 		ctx,
-		nil,
-		api.BypassMutableStateConsistencyPredicate,
-		workflowKey,
+		namespace.ID(workflowKey.NamespaceID),
+		commonpb.WorkflowExecution{
+			WorkflowId: workflowKey.WorkflowID,
+			RunId:      workflowKey.RunID,
+		},
 		workflow.LockPriorityHigh,
 	)
 	if err != nil {
 		return err
 	}
 	defer func() {
-		wfContext.GetReleaseFn()(retError)
-		wfContext.GetContext().Clear()
+		releaseFn(retError)
+		wfContext.Clear()
 	}()
 
-	mutableState := wfContext.GetMutableState()
-	_, dbRecordVersion := mutableState.GetUpdateCondition()
+	rebuildMutableState, err := r.replayNewWorkflow(
+		ctx,
+		workflowKey,
+		branchToken,
+	)
+	if err != nil {
+		return err
+	}
+	return r.writeToDB(ctx, rebuildMutableState)
+}
 
-	requestID := mutableState.GetExecutionState().CreateRequestId
-	versionHistories := mutableState.GetExecutionInfo().VersionHistories
-	currentVersionHistory, err := versionhistory.GetCurrentVersionHistory(versionHistories)
+func (r *workflowRebuilderImpl) writeToDB(
```

Comment:
what is the diff to overwriteToDB()

Reply:
since import workflow execution API is landed, i will remove this field / functionality